velox/exec/HashProbe.cpp

```diff
@@ -189,58 +225,276 @@ void HashProbe::initializeFilter(
   filterInputType_ = ROW(std::move(names), std::move(types));
 }
 
-BlockingReason HashProbe::isBlocked(ContinueFuture* future) {
-  if (table_) {
-    return BlockingReason::kNotBlocked;
+void HashProbe::maybeSetupSpillInput(
+    const std::optional<SpillPartitionId>& restoredPartitionId,
+    const SpillPartitionIdSet& spillPartitionIds) {
+  VELOX_CHECK_NULL(spillInputReader_);
+
+  // If 'restoredPartitionId' is not null, then 'table_' is built from the
+  // spilled build data. We need to create an unsorted reader to read the probe
+  // inputs from the corresponding spilled probe partition on disk.
+  if (restoredPartitionId.has_value()) {
+    auto iter = spillPartitionSet_.find(restoredPartitionId.value());
+    VELOX_CHECK(iter != spillPartitionSet_.end());
+    auto partition = std::move(iter->second);
+    VELOX_CHECK_EQ(partition->id(), restoredPartitionId.value());
+    spillInputReader_ = partition->createReader();
+    spillPartitionSet_.erase(iter);
   }
 
-  auto hashBuildResult =
-      operatorCtx_->task()
-          ->getHashJoinBridge(
-              operatorCtx_->driverCtx()->splitGroupId, planNodeId())
-          ->tableOrFuture(future);
+  VELOX_CHECK_NULL(spiller_);
```

Comment:
nit: would be nice to add a comment for this block of code to clarify that previous blocks sets up reading previously spilled partition while this block sets up spilling for the incoming data.